### ISSUE TITLE
[BUG] Some of TimeSeriesPredictor algorithms cause kernel restarting in running .fit() 

### ISSUE THREAD

**Issue by ttsuzuku** [autogluon/autogluon#4506]
**Bug Report Checklist**

- [x] I provided code that demonstrates a minimal reproducible example.  
- [x] I confirmed bug exists on the latest mainline of AutoGluon via source install.  
- [x] I confirmed bug exists on the latest stable version of AutoGluon.  

**Describe the bug**

In AG1.1.1, TimeSeriesPredictor.fit() failed with some of timeseries algorithms. JupyterLab kernel is restarting with no error message.

List of methods : [Forecasting Time Series - Model Zoo](https://auto.gluon.ai/stable/tutorials/timeseries/forecasting-model-zoo.html#forecasting-model-zoo)

- NG (JupyterLab kernel restarting with no error message) : DeepAR, DLinear, PatchTST, SimpleFeedForward, TemporalFusionTransformer, TiDE, WaveNet, Chronos
- OK : Naive, SeasonalNaive, Average, SeasonalAverage, Zero, ETS, AutoARIMA, AutoETS, AutoCES, Theta, ADIDA, CrostonSBA, IMAPA, NPTS, DirectTabular, RecursiveTabular, DynamicOptimizedTheta

(*) CrostonClassic and CrostonOptimize were not available in AG1.1.1.

**Expected behavior**

All methods I listed above work .fit() with no error.

**To Reproduce**

Environment: 
- AWS SageMaker Studio
- Image : SageMaker Distribution 2.0 (including AG1.1.1 by default)
- Instance type : ml.m5.2xlarge, ml.m5.4xlarge, ml.p3.8xlarge (withGPU)

Reproducing code:

```
import pandas as pd
df = pd.read_csv("https://autogluon.s3.amazonaws.com/datasets/timeseries/m4_hourly_subset/train.csv")
from autogluon.timeseries import TimeSeriesDataFrame, TimeSeriesPredictor
data = TimeSeriesDataFrame(df)

methods = ['Naive','SeasonalNaive', 'Average', 'SeasonalAverage', 'Zero', 'ETS', 'AutoARIMA', 'AutoETS', 'AutoCES', 'Theta', 'ADIDA', 
           'CrostonClassic', 'CrostonSBA', 'IMAPA', 'NPTS', 'DeepAR', 'DLinear', 'PatchTST', 'SimpleFeedForward', 'TemporalFusionTransformer', 
           'TiDE', 'WaveNet', 'DirectTabular', 'RecursiveTabular', 'DynamicOptimizedTheta', 'Chronos']

for method in methods:
    print('#'*50)
    print(method)
    predictor = TimeSeriesPredictor().fit(
        data, 
        hyperparameters={
            method:{},
        }
    )
```

**Screenshots / Logs**

NG: kernel restarting popup message
```
Kernel Restarting
The kernel for xxxx/xxxx.ipynb appears to have died. It will restart automatically.
```

OK: (ex: Naive)
```
Beginning AutoGluon training...
AutoGluon will save models to 'AutogluonModels[/ag-20240930_133752](https://xxxxxxxxxxxx.studio.ap-northeast-1.sagemaker.aws/ag-20240930_133752)'
=================== System Info ===================
AutoGluon Version:  1.1.1
Python Version:     3.11.9
Operating System:   Linux
Platform Machine:   x86_64
Platform Version:   #1 SMP Tue Sep 10 22:02:55 UTC 2024
CPU Count:          8
GPU Count:          0
Memory Avail:       26.54 GB [/](https://xxxxxxxxxxxx.studio.ap-northeast-1.sagemaker.aws/) 30.89 GB (85.9%)
Disk Space Avail:   93.27 GB [/](https://xxxxxxxxxxxx.studio.ap-northeast-1.sagemaker.aws/) 99.95 GB (93.3%)
===================================================

Fitting with arguments:
{'enable_ensemble': True,
 'eval_metric': WQL,
 'hyperparameters': {'Naive': {}},
 'known_covariates_names': [],
 'num_val_windows': 1,
 'prediction_length': 1,
 'quantile_levels': [0.1, 0.2, 0.3, 0.4, 0.5, 0.6, 0.7, 0.8, 0.9],
 'random_seed': 123,
 'refit_every_n_windows': 1,
 'refit_full': False,
 'skip_model_selection': False,
 'target': 'target',
 'verbosity': 2}

Inferred time series frequency: 'h'
Provided train_data has 148060 rows, 200 time series. Median time series length is 700 (min=700, max=960). 

Provided data contains following columns:
	target: 'target'

AutoGluon will gauge predictive performance using evaluation metric: 'WQL'
	This metric's sign has been flipped to adhere to being higher_is_better. The metric score can be multiplied by -1 to get the metric value.
===================================================

Starting training. Start time is 2024-09-30 13:37:52
Models that will be trained: ['Naive']
Training timeseries model Naive. 
	-0.0379       = Validation score (-WQL)
	0.13    s     = Training runtime
	1.54    s     = Validation (prediction) runtime
Not fitting ensemble as only 1 model was trained.
Training complete. Models trained: ['Naive']
Total runtime: 1.68 s
Best model: Naive
Best model score: -0.0379
```

**Installed Versions**

<details>

```python
# Replace this code with the output of the following:
from autogluon.core.utils import show_versions
show_versions()

INSTALLED VERSIONS
------------------
date                   : 2024-10-01
time                   : 02:37:39.428144
python                 : 3.11.9.final.0
OS                     : Linux
OS-release             : 5.10.225-213.878.amzn2.x86_64
Version                : #1 SMP Tue Sep 10 22:02:55 UTC 2024
machine                : x86_64
processor              : x86_64
num_cores              : 16
cpu_ram_mb             : 62825.47265625
cuda version           : None
num_gpus               : 0
gpu_ram_mb             : []
avail_disk_size_mb     : 95235

accelerate             : 0.21.0
autogluon              : 1.1.1
autogluon.common       : 1.1.1
autogluon.core         : 1.1.1
autogluon.features     : 1.1.1
autogluon.multimodal   : 1.1.1
autogluon.tabular      : 1.1.1
autogluon.timeseries   : 1.1.1
boto3                  : 1.34.131
catboost               : 1.2.5
defusedxml             : 0.7.1
evaluate               : 0.4.1
fastai                 : 2.7.16
gluonts                : 0.14.3
hyperopt               : 0.2.7
imodels                : None
jinja2                 : 3.1.4
joblib                 : 1.4.2
jsonschema             : 4.21.1
lightgbm               : 4.3.0
lightning              : 2.3.3
matplotlib             : 3.9.2
mlforecast             : 0.10.0
networkx               : 3.3
nlpaug                 : 1.1.11
nltk                   : 3.9.1
nptyping               : 2.4.1
numpy                  : 1.26.4
nvidia-ml-py3          : None
omegaconf              : 2.3.0
onnxruntime-gpu        : None
openmim                : 0.3.7
optimum                : None
optimum-intel          : None
orjson                 : 3.10.7
pandas                 : 2.2.2
pdf2image              : 1.17.0
Pillow                 : 10.4.0
psutil                 : 5.9.8
pytesseract            : 0.3.10
pytorch-lightning      : 2.3.3
pytorch-metric-learning: 2.3.0
ray                    : 2.31.0
requests               : 2.32.3
scikit-image           : 0.20.0
scikit-learn           : 1.4.2
scikit-learn-intelex   : None
scipy                  : 1.12.0
seqeval                : 1.2.2
setuptools             : 72.1.0
skl2onnx               : None
statsforecast          : 1.4.0
tabpfn                 : None
tensorboard            : 2.17.1
text-unidecode         : 1.3
timm                   : 0.9.16
torch                  : 2.3.1.post100
torchmetrics           : 1.2.1
torchvision            : 0.18.1a0+405940f
tqdm                   : 4.66.5
transformers           : 4.38.2
utilsforecast          : 0.0.10
vowpalwabbit           : None
xgboost                : 2.0.3
```

</details>

**Comment by shchur** [comment]
Hi @ttsuzuku, the problem likely comes from conflicts among packages in the conda environment. Here are two possible ways to fix this:
- Downgrade tensorflow: `micromamba install -c conda-forge tensorflow==2.15`
- Remove tensorflow completely: `micromamba remove tensorflow`

**Comment by ttsuzuku** [comment]
> Hi @ttsuzuku, the problem likely comes from conflicts among packages in the conda environment. Here are two possible ways to fix this:
> 
> * Downgrade tensorflow: `micromamba install -c conda-forge tensorflow==2.15`
> * Remove tensorflow completely: `micromamba remove tensorflow`

@shchur Thank you. It worked!!

I tried downgrading tensorflow as follows,

```
micromamba install -y -c conda-forge tensorflow==2.15
``` 
After restarting kernel, I could run DeepAR, DLinear, PatchTST, SimpleFeedForward, TemporalFusionTransformer, TiDE, WaveNet and Chronos successfully. It seems my issue has been fixed.

**FYI: Installed Versions (after downgrading tensorflow to 2.15**

<details>

```python
INSTALLED VERSIONS
------------------
date                   : 2024-10-02
time                   : 00:57:23.771859
python                 : 3.11.9.final.0
OS                     : Linux
OS-release             : 5.10.225-213.878.amzn2.x86_64
Version                : #1 SMP Tue Sep 10 22:02:55 UTC 2024
machine                : x86_64
processor              : x86_64
num_cores              : 8
cpu_ram_mb             : 31627.4296875
cuda version           : None
num_gpus               : 0
gpu_ram_mb             : []
avail_disk_size_mb     : 91864

accelerate             : 0.21.0
autogluon              : 1.1.1
autogluon.common       : 1.1.1
autogluon.core         : 1.1.1
autogluon.features     : 1.1.1
autogluon.multimodal   : 1.1.1
autogluon.tabular      : 1.1.1
autogluon.timeseries   : 1.1.1
boto3                  : 1.34.131
catboost               : 1.2.5
defusedxml             : 0.7.1
evaluate               : 0.4.1
fastai                 : 2.7.16
gluonts                : 0.14.3
hyperopt               : 0.2.7
imodels                : None
jinja2                 : 3.1.4
joblib                 : 1.4.2
jsonschema             : 4.23.0
lightgbm               : 4.3.0
lightning              : 2.3.3
matplotlib             : 3.9.2
mlforecast             : 0.10.0
networkx               : 3.3
nlpaug                 : 1.1.11
nltk                   : 3.9.1
nptyping               : 2.4.1
numpy                  : 1.26.4
nvidia-ml-py3          : None
omegaconf              : 2.3.0
onnxruntime-gpu        : None
openmim                : 0.3.7
optimum                : None
optimum-intel          : None
orjson                 : 3.10.7
pandas                 : 2.2.2
pdf2image              : 1.17.0
Pillow                 : 10.4.0
psutil                 : 5.9.8
pytesseract            : 0.3.10
pytorch-lightning      : 2.3.3
pytorch-metric-learning: 2.3.0
ray                    : 2.24.0
requests               : 2.32.3
scikit-image           : 0.20.0
scikit-learn           : 1.4.2
scikit-learn-intelex   : None
scipy                  : 1.12.0
seqeval                : 1.2.2
setuptools             : 72.1.0
skl2onnx               : None
statsforecast          : 1.4.0
tabpfn                 : None
tensorboard            : 2.15.2
text-unidecode         : 1.3
timm                   : 0.9.16
torch                  : 2.1.2.post100
torchmetrics           : 1.2.1
torchvision            : 0.16.1+b88453f
tqdm                   : 4.66.5
transformers           : 4.38.2
utilsforecast          : 0.0.10
vowpalwabbit           : None
xgboost                : 2.0.3
</details>

**Comment by push44** [comment]
I am still facing this issue and Tensorflow is not even installed. I using my Macbook pro. Installed Autolguon version is 1.4.0. Attached screenshot for reference. Any help is greatly appreciated. 

<img width="1837" height="1236" alt="Image" src="https://github.com/user-attachments/assets/cf0cdb5f-006f-4204-8b65-1ef9dcd50c4f" />
